### PR TITLE
Add browser-based tests, and other small cleanup

### PR DESCRIPTION
**Pull request by forevermatt**
### Added
- Add test to ensure status check returns OK
- Check the initial login page (following the redirect) during tests

### Fixed
- Switch to new PHP inspection engine on Scrutinizer
- Fix directions on how to do manual testing (in README file)
- Stop trying to use auth.json to provide GitHub credential
- Switch to official MariaDB image for test databases
- Fix namespace for our RequestTest class
- Use asset-packagist (in preparation for upgrading to Composer 2 later)

---

@Schparky and @jason-jackson, here's a re-submission of the remaining changes in this feature branch. Much of this (but not all of it) is code you've already seen. Sorry for the redo.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,6 +17,9 @@ behatappend:
 behatv:
 	docker-compose run --rm tests bash -c "vendor/bin/behat --config=features/behat.yml --strict --stop-on-failure -v"
 
+behatwip:
+	docker-compose run --rm tests bash -c "vendor/bin/behat --config=features/behat.yml --strict --stop-on-failure --tags '@wip'"
```
Comment (briskt) [inline Makefile:21]: I don't see the `wip` tag referenced. Is this no longer needed, or something I missed?

File: codeship-services.yml [hunk 2]
```diff
@@ -24,3 +25,68 @@ web:
...
+      EMAIL_SERVICE_baseUrl: "dummy"
+      EMAILER_CLASS: Sil\SilIdBroker\Behat\Context\fakes\FakeEmailer
+      HELP_CENTER_URL: "https://example.org/help"
+      PASSWORD_FORGOT_URL: "https://example.org/forgot"
+      PASSWORD_PROFILE_URL: "https://example.org/profile"
+      SUPPORT_EMAIL: "support@example.org"
+      EMAIL_SIGNATURE: "one red pill, please"
+      API_ACCESS_KEYS: "test-cli-abc123"
+      APP_ENV: "prod"
+      MIGRATE_PW_FROM_LDAP: "false"
```
Comment (briskt) [inline codeship-services.yml]: `MIGRATE_PW_FROM_LDAP` is obsolete and can be removed

File: codeship-services.yml [hunk 3]
```diff
@@ -24,3 +25,68 @@ web:
...
+      EMAIL_SERVICE_assertValidIp: "false"
+      EMAIL_SERVICE_baseUrl: "dummy"
+      EMAILER_CLASS: Sil\SilIdBroker\Behat\Context\fakes\FakeEmailer
+      HELP_CENTER_URL: "https://example.org/help"
+      PASSWORD_FORGOT_URL: "https://example.org/forgot"
+      PASSWORD_PROFILE_URL: "https://example.org/profile"
+      SUPPORT_EMAIL: "support@example.org"
+      EMAIL_SIGNATURE: "one red pill, please"
+      API_ACCESS_KEYS: "test-cli-abc123"
+      APP_ENV: "prod"
```
Comment (briskt) [inline codeship-services.yml:78]: `APP_ENV` defaults to "prod" so this can be removed if you want.

**Review by briskt — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (forevermatt) [reply Makefile:21]: It's for temporarily adding to a Behat Feature or Scenario while working on it, so it's easier to run just that test during development. For example:
```gherkin
Feature: Does something awesome

  @wip
  Scenario: Handles basic awesomeness requirements
    ... 
```

It's not intended to be committed into the code, which is why you don't see that Work-In-Progress tag (`@wip`) anywhere else.

**Pull request merged**